merlin/systems/triton/__init__.py

```diff
@@ -150,9 +150,6 @@ def _convert_tensor(t):
     out = t.as_numpy()
     if len(out.shape) == 2:
         out = out[:, 0]
-    # cudf doesn't seem to handle dtypes like |S15 or object that well
-    if is_string_dtype(out.dtype):
-        out = out.astype("str")
```

Comment:
My impression is that this does (or did) cover a real edge case, which may not be adequately covered by tests. This piece of code was inherited from the old serving code in NVT, which was TBH not very well tested.

Reply:
I've updated this keeing the string type coercion. Using `np.char.decode(out.astype(bytes))` intead of `out.astype("str")`.

It appears we do need this because cudf doesn't accept an array of byte strings as a type when constructing a DataFrame.